src/flamenco/runtime/tests/fd_exec_instr_test.c

```diff
@@ -1789,9 +1790,20 @@ fd_exec_vm_syscall_test_run( fd_exec_instr_test_runner_t * runner,
 
   /* Actually invoke the syscall */
   int syscall_err = syscall->func( vm, vm->reg[1], vm->reg[2], vm->reg[3], vm->reg[4], vm->reg[5], &vm->reg[0] );
+  if( syscall_err ) {
+    fd_log_collector_program_failure( vm->instr_ctx );
+  }
 
   /* Capture the effects */
-  effects->error = -syscall_err;
+  int exec_err = vm->instr_ctx->exec_err;
+  effects->error = 0;
+  if( syscall_err ) {
+    effects->error = exec_err <= 0 ? -vm->instr_ctx->exec_err : -1;
+    if( exec_err==0 ) {
+      FD_LOG_WARNING(( "TODO: syscall returns error, but exec_err not set. this is probably missing a log." ));
```

Comment:
wondering if this will get too spammy.. maybe it should be NOTICE? after we've fixed all the syscalls we can flip it back to WARNING.

Reply:
For the fixtures we have, we never hit that. So it should be pretty silent. This is only an edge case where we return an error but we don't track Agave's corresponding error. I've already fixed all crypto + logging, including the underlying macros like CU, memory mapping.